examples/basic/ipc_shared_memory_server.c

```diff
@@ -0,0 +1,196 @@
+             sizeof(server_addr)) < 0) {
+        fprintf(stderr, "[server] ERROR: cannot bind to the port\n");
+        goto err_close_server_socket;
+    }
+
+    fprintf(stderr, "[server] Binding done\n");
+
+    // listen for the client
+    if (listen(server_socket, 1) < 0) {
+        fprintf(stderr, "[server] ERROR: listen() failed\n");
+        goto err_close_server_socket;
+    }
+
+    fprintf(stderr, "[server] Listening for incoming connections ...\n");
+
+    // accept an incoming connection
+    client_addr_len = sizeof(client_addr);
+    client_socket = accept(server_socket, (struct sockaddr *)&client_addr,
+                           (socklen_t *)&client_addr_len);
+    if (client_socket < 0) {
+        fprintf(stderr, "[server] ERROR: accept() failed\n");
+        goto err_close_server_socket;
+    }
+
+    fprintf(stderr, "[server] Client connected at IP %s and port %i\n",
+            inet_ntoa(client_addr.sin_addr), ntohs(client_addr.sin_port));
+
+    // zero the IPC_handle buffer
+    memset(IPC_handle, 0, IPC_handle_size);
+
+    // receive a client's message
+    if (recv(client_socket, IPC_handle, IPC_handle_size, 0) < 0) {
+        fprintf(stderr, "[server] ERROR: recv() failed\n");
+        goto err_close_client_socket;
+    }
+
+    fprintf(stderr, "[server] Received an IPC handle from the client\n");
+
+    void *SHM_ptr;
+    umf_result = umfMemoryProviderOpenIPCHandle(OS_memory_provider, IPC_handle,
+                                                &SHM_ptr);
+    if (umf_result != UMF_RESULT_SUCCESS) {
+        fprintf(stderr, "[server] ERROR: opening the IPC handle failed\n");
+        goto err_close_client_socket;
+    }
+
+    fprintf(stderr,
+            "[server] Opened the IPC handle received from the client\n");
+
+    // read the current value from the shared memory
+    unsigned long long SHM_number_1 = *(unsigned long long *)SHM_ptr;
+    // calculate the new value
+    unsigned long long SHM_number_2 = SHM_number_1 / 2;
+    // write the new number directly to the client's shared memory
+    *(unsigned long long *)SHM_ptr = SHM_number_2;
+
+    fprintf(
+        stderr,
+        "[server] Wrote a new number directly to the client's shared memory\n");
+
+    // write the response to the server_message buffer
+    memset(server_message, 0, sizeof(server_message));
+    strcpy(server_message, SERVER_MSG);
+
+    // send response to the client
+    if (send(client_socket, server_message, strlen(server_message), 0) < 0) {
+        fprintf(stderr, "[server] ERROR: send() failed\n");
+        goto err_CloseIPCHandle;
+    }
+
+    fprintf(stderr, "[server] Sent response to the client\n");
+
+    ret = 0; // SUCCESS
+
+err_CloseIPCHandle:
+    // we do NOT know the exact size of the remote shared memory! - TODO: the API should be changed
+    umf_result = umfMemoryProviderCloseIPCHandle(OS_memory_provider, SHM_ptr,
+                                                 4096 /* TODO: change API */);
```

Comment:
@vinser52 @bratpiorka I think the API should be changed here in this way:
https://github.com/ldorau/unified-memory-framework/commit/de9a1ab4e24ef958643f798ed65703869517c2e6